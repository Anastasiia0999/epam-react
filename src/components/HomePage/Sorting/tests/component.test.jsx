import React from 'react';
import { mount} from 'enzyme';

import {Sorting} from "../index";

describe('<Sorting />', () =>{
    let wrapper;
    let instance;

    const props = {
        sortByLikesDispatch: jest.fn(),
        sortByRatingDispatch: jest.fn(),
        resetDispatch: jest.fn(),
        fields:{
            "sorting-title" : "Sort movies by",
            "likes-btn": "Likes",
            "stars-btn": "Rating",
            "reset-btn": "Reset"
        }
    };

    beforeEach(() => {
        wrapper = mount(<Sorting {...props} />);
        instance = wrapper.instance();
    });

    it('should render correctly', () =>{
        expect(wrapper.html()).toMatchSnapshot();
    });

});


