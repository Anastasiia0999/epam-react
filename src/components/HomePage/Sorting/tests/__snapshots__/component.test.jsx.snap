// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sorting /> should render correctly 1`] = `"<div class=\\"card sorting col-12\\"><div class=\\"card-body sort-box\\"><h4>Sort movies by</h4><hr><button type=\\"button\\" class=\\"btn  button-sort\\">Likes</button><button type=\\"button\\" class=\\"btn  button-sort\\">Rating</button><button type=\\"button\\" class=\\"btn  button-sort\\">Reset</button></div></div>"`;
